pubtools/pulplib/_impl/model/repository/container.py

```diff
@@ -47,7 +47,8 @@ def _data_to_init_args(cls, data):
 
         for dist in data.get("distributors") or []:
             if dist["distributor_type_id"] == "docker_distributor_web":
-                out["registry_id"] = dist["config"].get("repo-registry-id")
+                if "repo-registry-id" in dist["config"]:
```

Comment:
I checked code in pulp_docker just now and it seems that, if repo-registry-id is set to null or empty string in config, then the default behavior of using the repo id also applies.  But this logic currently doesn't match that, it only uses the repo id if repo-registry-id is *absent*.  So I think this is still not quite matching Pulp logic, and it should instead be something like:

```suggestion
                registry_id = dist["config"].get("repo-registry-id")
                if registry_id:
                    out["registry_id"] = registry_id
```

Reply:
Thanks for pointing this out.